### DISCUSSION TITLE
How to call `day.js` function to get ISO week number?

### DISCUSSION THREAD

**Discussion by Norman-h** [stdword/logseq13-full-house-plugin#31] [Q&A] [answered]
I have template like this:
```
Today Work [[``{ date.from(c.page.name).year() }``-W``{ date.from(c.page.name).week() }``]]
```
But I need call day.js `isoweek`. How can I do? Thanks!

**Comment by stdword** [comment, marked as answer]
Hello!

### Just like `days.js` [documentation](https://day.js.org/docs/en/get-set/iso-week) shows:

<img width="250px" src="https://github.com/stdword/logseq13-full-house-plugin/assets/1984175/ac460333-41f6-4330-ad86-6dd04027c851"/>


### For your example I've made some changes:
1. Prefer new syntax style: ` ``...`` ` instead of old one: ` ``{...}`` `
2. Used `c.page.day` to access `day.js` object related to journal page (it is empty for non-journal), instead of parsing page name with `date.from`
3. Changed `.week` to `.isoWeek`
4. Add `Variant 2` to access iso week number via `.format` (see `day.js` [documentation](https://day.js.org/docs/en/plugin/advanced-format))
5. `Variant 2` also shows a way to get off the clickable ref inside template

```
Variant 1: Today Work [[``c.page.day.year()``-W``c.page.day.isoWeek()``]]
Variant 2: Today Work ``[c.page.day.format('YYYY-[W]W')]``
```

<img width="500px" src="https://github.com/stdword/logseq13-full-house-plugin/assets/1984175/9613b84d-1dd0-4918-9905-986d762b2b35"/>

**Reply by stdword** [reply]
Also note [this](https://github.com/stdword/logseq13-full-house-plugin/discussions/43) approach with **Weekly Pages**

**Comment by Norman-h** [comment]
Thanks!

**Comment by Norman-h** [comment]
@stdword Sorry, now I have week note, and template like:
```
week   #``{ date.now.year() }``-M``{ date.now.month() }`` #``{ date.now.year() }``
```
but now is October and I get Sep.
```
week #2023-M10 #2023  // I need
week #2023-M9 #2023   // I get
```

**Reply by stdword** [reply]
### Your template
Note `day.js`'s documentation for [`.month()`](https://day.js.org/docs/en/get-set/month). It says:
> Months are zero indexed, so January is month 0

So just add 1:
```
week   #``{ date.now.year() }``-M``{ date.now.month() + 1 }`` #``{ date.now.year() }``
```

### Another way
Is to use `day.js`'s [`.format()`](https://day.js.org/docs/en/display/format) (as I mentioned above):
```
week ``date.now.format('[#]YYYY-[M]M')`` ``date.now.format('[#]YYYY')``
```
